.github/workflows/unit_tests.yml

```diff
@@ -58,7 +58,7 @@ jobs:
           RUSTC_WORKSPACE_WRAPPER: sccache
         with:
           command: clippy
-          args: -- --no-deps -D warnings
+          args: --all-targets --all-features -- --no-deps -D warnings -A clippy::type_complexity
```

Comment:
Could we also have `type_complexity` checks? I know AlephBFT has this, but imo that's a mistake and it would be nice to fix it there as well.

Reply:
+1

Reply:
Jira issue added :)

Reply:
I have reworked this PR to include the complexity checks. The setup should now prevent us from submitting overly complex types.